app/assets/javascripts/discourse/app/lib/lock-on.js

```diff
@@ -41,7 +41,14 @@ export default class LockOn {
     }
 
     const { top } = element.getBoundingClientRect();
-    const offset = top + window.scrollY;
+    let offset = top + window.scrollY;
+    if (this.options.originalTopOffset) {
+      // if element's original top offset is in the bottom half of the viewport
+      // jump to it, otherwise respect the offset
+      if (window.innerHeight / 2.25 > this.options.originalTopOffset) {
```

Comment:
what is 2.25 here ? mays we use a constant with a good name to make it explicit ?

Reply:
I'm having a hard time coming up with a good name for this. In fact, it feels easier to understand to me what 2.25 is doing when it is in the line. You can see it is used to calculate whether the element's top offset is roughly above the viewport's middle line. (I.e. viewport height divided by two but with a small adjustment to account for the header.)